tests/spark_autologging/test_pyspark_ml_autologging.py

```diff
@@ -0,0 +1,237 @@
+import importlib
+import pytest
+from collections import namedtuple
+from distutils.version import LooseVersion
+from unittest import mock
+
+import mlflow
+from mlflow.utils import truncate_dict
+from mlflow.utils.validation import (
+    MAX_PARAM_VAL_LENGTH,
+    MAX_ENTITY_KEY_LENGTH,
+)
+
+import pyspark
+from pyspark.sql import SparkSession
+from pyspark.ml.linalg import Vectors
+from pyspark.ml.regression import LinearRegression, LinearRegressionModel
+from mlflow.pyspark.ml import (
+    _should_log_model,
+    _get_instance_param_map,
+    _get_warning_msg_for_skip_log_model,
+    _get_warning_msg_for_fit_call_with_a_list_of_params,
+)
+
+MODEL_DIR = "model"
+MLFLOW_PARENT_RUN_ID = "mlflow.parentRunId"
+
+
+@pytest.fixture(scope="module")
+def spark_session():
+    session = SparkSession.builder.master("local[*]").getOrCreate()
+    yield session
+    session.stop()
+
+
+@pytest.fixture(scope="module")
+def dataset_binomial(spark_session):
+    return spark_session.createDataFrame(
+        [(1.0, Vectors.dense(1.0)), (0.0, Vectors.sparse(1, [], []))] * 100, ["label", "features"]
+    )
+
+
+@pytest.fixture(scope="module")
+def dataset_multinomial(spark_session):
+    return spark_session.createDataFrame(
+        [(1.0, Vectors.dense(1.0)), (0.0, Vectors.sparse(1, [], [])), (2.0, Vectors.dense(0.5))]
+        * 100,
+        ["label", "features"],
+    )
+
+
+def truncate_param_dict(d):
+    return truncate_dict(d, MAX_ENTITY_KEY_LENGTH, MAX_PARAM_VAL_LENGTH)
+
+
+def stringify_dict_values(d):
+    return {k: str(v) for k, v in d.items()}
+
+
+def get_expected_class_tags(estimator):
+    return {
+        "estimator_name": estimator.__class__.__name__,
+        "estimator_class": estimator.__class__.__module__ + "." + estimator.__class__.__name__,
+    }
+
+
+def get_run_data(run_id):
+    client = mlflow.tracking.MlflowClient()
+    data = client.get_run(run_id).data
+    # Ignore tags mlflow logs by default (e.g. "mlflow.user")
+    tags = {k: v for k, v in data.tags.items() if not k.startswith("mlflow.")}
+    artifacts = [f.path for f in client.list_artifacts(run_id)]
+
+    RunData = namedtuple("RunData", ["params", "metrics", "tags", "artifacts"])
+    return RunData(data.params, data.metrics, tags, artifacts)
+
+
+def load_model_by_run_id(run_id):
+    return mlflow.spark.load_model("runs:/{}/{}".format(run_id, MODEL_DIR))
+
+
+def get_dataset_binomial(spark_session):
+    return spark_session.createDataFrame(
+        [(1.0, Vectors.dense(1.0)), (0.0, Vectors.sparse(1, [], []))] * 100, ["label", "features"]
+    )
+
+
+def test_basic_estimator(dataset_binomial):
```

Comment:
@harupy 
Shall we mark these test by `pytest.mark.large` like other spark autologging tests ?

Reply:
Yes :)